target_chains/ethereum/contracts/contracts/pyth/PythAssetRegistry.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+import "@pythnetwork/pyth-sdk-solidity/IPyth.sol";
+
+error InconsistentParamsLength();
+
+contract PythAssetRegistryStorage {
+    struct State {
+        address pyth;
+        address BASE_CURRENCY;
+        uint256 BASE_CURRENCY_UNIT;
+        // Map of asset priceIds (asset => priceId)
+        mapping(address => bytes32) assetsPriceIds;
+    }
+}
+
+contract PythAssetRegistry {
+    PythAssetRegistryStorage.State _registryState;
+
+    function pyth() public view returns (IPyth) {
+        return IPyth(_registryState.pyth);
+    }
+
+    /**
+     * @dev Emitted after the base currency is set
+     * @param baseCurrency The base currency of used for price quotes
+     * @param baseCurrencyUnit The unit of the base currency
+     */
+    event BaseCurrencySet(
+        address indexed baseCurrency,
+        uint256 baseCurrencyUnit
+    );
+
+    /**
+     * @dev Emitted after the price source of an asset is updated
+     * @param asset The address of the asset
+     * @param source The priceId of the asset
+     */
+    event AssetSourceUpdated(address indexed asset, bytes32 indexed source);
+
+    function setPyth(address pythAddress) internal {
+        _registryState.pyth = payable(pythAddress);
+    }
+
+    function setAssetsSources(
+        address[] memory assets,
+        bytes32[] memory priceIds
+    ) internal {
+        if (assets.length != priceIds.length) {
+            revert InconsistentParamsLength();
+        }
+        for (uint256 i = 0; i < assets.length; i++) {
+            _registryState.assetsPriceIds[assets[i]] = priceIds[i];
+            emit AssetSourceUpdated(assets[i], priceIds[i]);
+        }
```

Comment:
this function name sounds like it replaces all existing keys in the map, but it doesn't. Maybe `insertAssetSources` ?

Reply:
typically would agree, but since we're planning on deploying this as an immutable contract & only calling `setAssetSources` once when initializing the contract I think it would be better to leave the name as-is. It also follows the Aave convention. Can still update the naming if you feel strongly a bout this